File: VB/WinSample.Module/MyReportParametersObject.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.Persistent.Base
Imports DevExpress.ExpressApp.Reports
Imports DevExpress.Xpo.DB
Imports DevExpress.ExpressApp

Namespace WinSample.Module
    <NonPersistent()> _
    Public Class MyReportParametersObject
        Inherits ReportParametersObjectBase
        Public Sub New(ByVal objectSpace As IObjectSpace, ByVal reportDataType As Type)
            MyBase.New(objectSpace, reportDataType)
        End Sub
        Public Overrides Function GetCriteria() As DevExpress.Data.Filtering.CriteriaOperator
            Return Nothing
        End Function
        Public Overrides Function GetSorting() As SortProperty()
            Return New SortProperty() {New SortProperty("MasterName", SortingDirection.Ascending)}
        End Function
        Private _StartDate As DateTime
        <Index(0)> _
        Public Property StartDate() As DateTime
            Get
                Return _StartDate
            End Get
            Set(ByVal value As DateTime)
                _StartDate = value
            End Set
        End Property
        Private _EndDate As DateTime
        <Index(1)> _
        Public Property EndDate() As DateTime
            Get
                Return _EndDate
            End Get
            Set(ByVal value As DateTime)
                _EndDate = value
            End Set
        End Property
    End Class

End Namespace

File: CS/WinSample.Module/MyReportParametersObject.cs
using System;

using DevExpress.Xpo;

using DevExpress.Persistent.Base;
using DevExpress.ExpressApp.Reports;
using DevExpress.Xpo.DB;

namespace WinSample.Module {
    [NonPersistent]
    public class MyReportParametersObject : ReportParametersObjectBase {
        public MyReportParametersObject(Session session) : base(session) { }
        public override DevExpress.Data.Filtering.CriteriaOperator GetCriteria() {
            return null;
        }
        public override SortingCollection GetSorting() {
            SortingCollection sorting = new SortingCollection();
            sorting.Add(new SortProperty("MasterName", SortingDirection.Ascending));
            return sorting;
        }
        private DateTime _StartDate;
        [Index(0)]
        public DateTime StartDate {
            get { return _StartDate; }
            set { SetPropertyValue("StartDate", ref _StartDate, value); }
        }
        private DateTime _EndDate;
        [Index(1)]
        public DateTime EndDate {
            get { return _EndDate; }
            set { SetPropertyValue("EndDate", ref _EndDate, value); }
        }
    }

}
